addons/sms/tests/test_sms_template.py

```diff
@@ -106,3 +106,28 @@ def test_sms_template_rendering_unrestricted(self):
 
         body = sms_composer._prepare_body_values(self.partner)[self.partner.id]
         self.assertIn(self.partner.name, body, 'Template Editor should be able to write new Jinja code')
+
+@tagged("post_install")
+class TestSMSComposer(TransactionCase):
+
+    def test_sending_on_sale_order(self):
+        partner = self.env['res.partner'].create({'name': 'Test Partner'})
+        sale_order = self.env['sale.order'].create({
+            'partner_id': partner.id,
+        })
+
+        sms_composer = self.env['sms.composer'].create({
+            'body': 'Test body',
+            'composition_mode': 'comment',
+            'mass_force_send': False,
+            'mass_keep_log': True,
+            'number_field_name': False,
+            'numbers': False,
+            'recipient_single_number_itf': '+32494670905',
+            'res_id': sale_order.id,
+            'res_model': 'sale.order'
+        })
+        try:
+            sms_composer.action_send_sms()
+        except ValueError as e:
+            self.fail(f"{e}\nThe action must be done despite the empty `number_field_name` field")
```

Comment:
Maybe check the message is sent while we're at it, this way the test is more generally applicable.
(It's easy to check, you can look at all the other tests in that file)